src/compiler/builder.ts

```diff
@@ -438,8 +438,22 @@ namespace ts {
                     removeSemanticDiagnosticsOf(state, f.resolvedPath)
                 );
             }
+            // When file is added to affected file because of global file change, the signature will not be update

```

Comment:
I'm having some trouble understanding this comment.  I think it's saying that if there was a change affecting the global scope (as checked by the surrounding if-block?) and if that change was that `affectedFile` was newly added (we don't appear to be checking this, but maybe `updateShapeSignature` will no-op if that's not the case?), then we need to emit the signature (i.e. .d.ts file) for `affectedFile` because the code for handling a global change wont' do that automatically?

Reply:
Global file's signature update adds all files to affected file list without calculating their signature so this makes sure to update the signature for the affected file.

Reply:
How's this?
```ts
// When a change affects the global scope, all files are considered to be affected so, as an optimization, 
// signatures are not computed.  If the file is new no signature will ever have been computed, so we do
// so here.
```

I don't quite understand why this is only a problem for  new files - won't existing files have stale signatures?

Reply:
It is not new file specific.. Just easy to repro with that scenario.

Reply:
Oh.  Then why is it applied only to `affectedFile` and not to all files?  Is this called once for each?  Or maybe it is applied to all files and `affectedFile` is passed in for some other reason?

Reply:
From your updated comment, I understand it to be called for each affected file.  The new comment is much clearer - thanks!